### PR TITLE
[hold merge] Make node containers nonprivileged in iptables mode

### PR DESCRIPTION
**Pull request by mgleung**
## Description

Make all the containers in calico-node run nonprivileged while running with the iptables dataplane.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/node.go [hunk 1]
```diff
@@ -825,14 +827,23 @@ func (c *nodeComponent) cniEnvvars() []v1.EnvVar {
...
+		Name:      "calico-node",
+		Image:     c.nodeImage,
+		Resources: c.nodeResources(),
+		SecurityContext: &v1.SecurityContext{
+			Privileged: ptr.BoolToPtr(false),
+			RunAsUser:  ptr.Int64ToPtr(0),
+			Capabilities: &v1.Capabilities{
+				Add: []v1.Capability{
+					v1.Capability("NET_RAW"),
+					v1.Capability("NET_ADMIN"),
```
Comment (fasaxc) [inline pkg/render/node.go:839]: Kernel v5.8 adds CAP_BPF.  BPF mode also needs to adjust resource limits; probably needs CAP_SYS_RESOURCE?

Comment (mgleung) [reply pkg/render/node.go]: Did a bit of experimenting. It appears that we still need `privileged` permissions to write to `bpffs`. We can probably adjust the permissions required if we're able to change where `tc` writes to but a quick test of setting `TC_BPF_MNT` proved unsuccessful for now. We'll probably need to tackle a reduced set of permissions for BPF at a later time.

Comment (mgleung) [reply pkg/render/node.go]: After some more testing, it seems that it should be possible to run BPF as non-privileged with some of the newer kernel versions (5.8+). There are a few more issues around mounts that need to be worked out, but I don't think we should block the removal of `privileged` permissions on the case anymore. We can remove privileged on BPF later.

**Comment by tmjd** [comment]
If this PR needs to go into v1.19 for the up coming enterprise release then once this is merged to master you'll need to cherry-pick into the release-v1.19 branch and add the Milestone of v1.19.0 to that PR.

**Pull request closed without merging**